h2o-algos/src/main/java/hex/tree/TreeHandler.java

```diff
@@ -119,12 +220,16 @@ static TreeProperties convertSharedTreeSubgraph(final SharedTreeSubgraph sharedT
         treeprops._features[0] = sharedTreeSubgraph.rootNode.getColName();
         treeprops._nas[0] = getNaDirection(sharedTreeSubgraph.rootNode);
         treeprops.levels = new int[sharedTreeSubgraph.nodesArray.size()][];
+        treeprops._languageTreeRepresentation = getLanguageRepresentation(sharedTreeSubgraph);
+        treeprops._languagePathsRepresentations[0] = "Predicted value: " + sharedTreeSubgraph.rootNode.getPredValue();
+
 
         List<SharedTreeNode> nodesToTraverse = new ArrayList<>();
         nodesToTraverse.add(sharedTreeSubgraph.rootNode);
         append(treeprops._rightChildren, treeprops._leftChildren,
                 treeprops._descriptions, treeprops._thresholds, treeprops._features, treeprops._nas,
                 treeprops.levels, treeprops._predictions, nodesToTraverse, -1, false);
+        fillLanguagePathRepresentation(treeprops, getDomainValues(sharedTreeSubgraph));
```

Comment:
As far as I can see, why aren't the new descriptions filled as a part of the original recursion ? It should have everything required, only one new parameter with domain values extracted is required - the `getDomainValues(sharedTreeSubgraph)` can be done before `append` is called for the first time.

Am I missing something ?

Reply:
That was my original intention but in getConditionByIndex() I use not only current node index when accessing information in tree properties, but parentIndex and targetIndex. in append() there is a "pointer" index by which it is possible to set up tree infos of current node. I think there might be a way how to do this, but not entirely straightforward - would leave for next PR or put this PR on hold.